util/gpool/spmc/worker_loop_queue.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spmc
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/resourcemanager/pooltask"
+)
+
+var (
+	// errQueueIsFull will be returned when the worker queue is full.
+	errQueueIsFull = errors.New("the queue is full")
+
+	// errQueueIsReleased will be returned when trying to insert item to a released worker queue.
+	errQueueIsReleased = errors.New("the queue is released could not accept item anymore")
+)
+
+type loopQueue[T any, U any, C any, CT any, TF pooltask.Context[CT]] struct {
+	items  []*goWorker[T, U, C, CT, TF]
```

Comment:
Is it possible to be a general loop queue? For example, replacing the `goWorker[T,U,C,CT,TF]` with one generic parameter.

Reply:
Oops, I realized that it depends on the `.recycleTime` of the `goWorker`